core/src/main/java/org/infinispan/interceptors/impl/BaseStateTransferInterceptor.java

```diff
@@ -223,7 +225,7 @@ private Object handleReadCommandReturn(InvocationContext rCtx, VisitableCommand
       } else if (ce instanceof AllOwnersLostException) {
          if (trace)
             getLog().tracef("All owners for command %s have been lost.", cmd);
-         return null;
+         return rCommand.acceptVisitor(rCtx, LostDataVisitor.INSTANCE);
```

Comment:
Last night I didn't think so, but now I'm leaning towards putting all the checks in DI again:

https://issues.jboss.org/browse/ISPN-7956